ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -439,43 +352,223 @@ def test_function(
+    **all_as_kwargs_np,
+):
+    """
+    Test a function that consumes (or returns) arrays for the current backend by
+    comparing the result with numpy.
+
+    Parameters
+    ----------
+    input_dtypes
+        data types of the input arguments in order.
+    test_flags
+        FunctionTestFlags object that stores all testing flags, including:
+        num_positional_args, with_out, instance_method, as_variable,
+        native_arrays, container, gradient
+    fw
+        current backend (framework).
+    fn_name
+        name of the function to test.
+    rtol_
+        relative tolerance value.
+    atol_
+        absolute tolerance value.
+    test_values
+        if True, test for the correctness of the resulting values.
+    xs_grad_idxs
+        Indices of the input arrays to compute gradients with respect to. If None,
+        gradients are returned with respect to all input arrays. (Default value = None)
+    ret_grad_idxs
+        Indices of the returned arrays for which to return computed gradients. If None,
+        gradients are returned for all returned arrays. (Default value = None)
+    ground_truth_backend
+        Ground Truth Backend to compare the result-values.
+    on_device
+        The device on which to create arrays
+    return_flat_np_arrays
+        If test_values is False, this flag dictates whether the original returns are
+        returned, or whether the flattened numpy arrays are returned.
+    all_as_kwargs_np
+        input arguments to the function as keyword arguments.
+
+    Returns
+    -------
+    ret
+        optional, return value from the function
+    ret_gt
+        optional, return value from the Ground Truth function
+
+    Examples
+    --------
+    >>> input_dtypes = 'float64'
+    >>> as_variable_flags = False
+    >>> with_out = False
+    >>> num_positional_args = 0
+    >>> native_array_flags = False
+    >>> container_flags = False
+    >>> instance_method = False
+    >>> test_flags = FunctionTestFlags(num_positional_args, with_out,
+        instance_method,
+        as_variable,
+        native_arrays,
+        container_flags,
+        none)
+    >>> fw = "torch"
+    >>> fn_name = "abs"
+    >>> x = np.array([-1])
+    >>> test_function(input_dtypes, test_flags, fw, fn_name, x=x)
+
+    >>> input_dtypes = ['float64', 'float32']
+    >>> as_variable_flags = [False, True]
+    >>> with_out = False
+    >>> num_positional_args = 1
+    >>> native_array_flags = [True, False]
+    >>> container_flags = [False, False]
+    >>> instance_method = False
+    >>> test_flags = FunctionTestFlags(num_positional_args, with_out,
+        instance_method,
+        as_variable,
+        native_arrays,
+        container_flags,
+        none)
+    >>> fw = "numpy"
+    >>> fn_name = "add"
+    >>> x1 = np.array([1, 3, 4])
+    >>> x2 = np.array([-3, 15, 24])
+    >>> test_function(input_dtypes, test_flags, fw, fn_name, x1=x1, x2=x2)
+    """
+    fw = backend_to_test
```

Comment:
No need to use the alias, let's use `backend_to_test` to make things straight forward.